kaldb/src/test/java/com/slack/kaldb/server/ManagerApiGrpcTest.java

```diff
@@ -0,0 +1,409 @@
+import io.micrometer.core.instrument.simple.SimpleMeterRegistry;
+import java.time.Instant;
+import java.util.Collections;
+import java.util.List;
+import org.apache.curator.test.TestingServer;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+
+@SuppressWarnings("ResultOfMethodCallIgnored")
+public class ManagerApiGrpcTest {
+  @Rule public final GrpcCleanupRule grpcCleanup = new GrpcCleanupRule();
+
+  private TestingServer testingServer;
+  private MeterRegistry meterRegistry;
+
+  private MetadataStore metadataStore;
+  private ServiceMetadataStore serviceMetadataStore;
+  private ManagerApiServiceGrpc.ManagerApiServiceBlockingStub managerApiStub;
+
+  @Before
+  public void setUp() throws Exception {
+    Tracing.newBuilder().build();
+    meterRegistry = new SimpleMeterRegistry();
+    testingServer = new TestingServer();
+
+    KaldbConfigs.ZookeeperConfig zkConfig =
+        KaldbConfigs.ZookeeperConfig.newBuilder()
+            .setZkConnectString(testingServer.getConnectString())
+            .setZkPathPrefix("ManagerApiGrpcTest")
+            .setZkSessionTimeoutMs(30000)
+            .setZkConnectionTimeoutMs(30000)
+            .setSleepBetweenRetriesMs(1000)
+            .build();
+
+    metadataStore = ZookeeperMetadataStoreImpl.fromConfig(meterRegistry, zkConfig);
+    serviceMetadataStore = spy(new ServiceMetadataStore(metadataStore, true));
+
+    grpcCleanup.register(
+        InProcessServerBuilder.forName(this.getClass().toString())
+            .directExecutor()
+            .addService(new ManagerApiGrpc(serviceMetadataStore))
+            .build()
+            .start());
+    ManagedChannel channel =
+        grpcCleanup.register(
+            InProcessChannelBuilder.forName(this.getClass().toString()).directExecutor().build());
+
+    managerApiStub = ManagerApiServiceGrpc.newBlockingStub(channel);
+  }
+
+  @After
+  public void tearDown() throws Exception {
+    serviceMetadataStore.close();
+    metadataStore.close();
+
+    testingServer.close();
+    meterRegistry.close();
+  }
+
+  @Test
+  public void shouldCreateAndGetNewService() {
+    String serviceName = "testService";
+    String serviceOwner = "testOwner";
+
+    managerApiStub.createServiceMetadata(
+        ManagerApi.CreateServiceMetadataRequest.newBuilder()
+            .setName(serviceName)
+            .setOwner(serviceOwner)
+            .build());
+
+    Metadata.ServiceMetadata getServiceMetadataResponse =
+        managerApiStub.getServiceMetadata(
+            ManagerApi.GetServiceMetadataRequest.newBuilder().setName(serviceName).build());
+    assertThat(getServiceMetadataResponse.getName()).isEqualTo(serviceName);
+    assertThat(getServiceMetadataResponse.getOwner()).isEqualTo(serviceOwner);
+    assertThat(getServiceMetadataResponse.getThroughputBytes()).isEqualTo(0);
+    assertThat(getServiceMetadataResponse.getPartitionConfigsList().size()).isEqualTo(0);
```

Comment:
Please add an assert reading this data from the servicemetadata store also. All the CRUD tests can benefit from these additional checks.

Reply:
Done. Added service metadata store checks to every method in this test suite.